codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -77,6 +84,17 @@ public DefaultHttp2ConnectionDecoder(Http2Connection connection,
         connection.local().flowController().frameWriter(encoder.frameWriter());
     }
 
+    public DefaultHttp2ConnectionDecoder(Http2Connection connection,
+                                         Http2ConnectionEncoder encoder,
+                                         Http2FrameReader frameReader,
+                                         Http2Settings initialSettings,
+                                         Http2PromisedRequestVerifier requestVerifier) throws Http2Exception {
+        this(connection, encoder, frameReader, requestVerifier);
+        if (initialSettings != null) {
+            applyLocalSettings(initialSettings);
```

Comment:
The responsibility for retaining the settings values and enforcing the settings constraints is spread out in different areas of the code and may be initialized with different values than the default specified in the RFC. It can be difficult to know how to configure all the places where the settings are enforced, so this `Http2Settings` object allows the initial settings to be configured consistently in a single place, and ensure the initial internal settings are also communicated to the peer.

IIUC a peer is allowed to choose w/e initial values they want (as eluded to in 3.5).

https://tools.ietf.org/html/rfc7540#section-3.5
> To avoid unnecessary latency, clients are permitted to send
   additional frames to the server immediately after sending the client
   connection preface, without waiting to receive the server connection
   preface.  It is important to note, however, that the server
   connection preface SETTINGS frame might include parameters that
   necessarily alter how a client is expected to communicate with the
   server